matminer/featurizers/composition.py

```diff
@@ -254,6 +255,83 @@ def citations(self):
     def implementors(self):
         return ["Jiming Chen", "Logan Ward", "Anubhav Jain", "Alex Dunn"]
 
+class Meredig(BaseFeaturizer):
+    """
+    Class to calculate features as defined in Meredig et. al.
+    """
+
+    def __init__(self):
+        self.data_source = MeredigData()
+
+        self.features = [e.strip()+" fraction" for e in self.data_source.element_names]
+        self.features = self.features+["mean AtomicWeight", "mean Column", "mean Row", "range AtomicNumber", "mean AtomicNumber",
+        "range AtomicRadius", "mean AtomicRadius", "range Electronegativity", "mean Electronegativity", "mean NsValence",
+        "mean NpValence", "mean NdValence", "mean NfValence", "fraction NsValence",
+        "fraction NpValence", "fraction NdValence", "fraction NfValence"]
+        # Initialize stats computer
+        self.pstats = PropertyStats()
+
+    def featurize(self, comp):
+        """
+        Get elemental property attributes
+
+        Args:
+            comp: Pymatgen composition object
+
+        Returns:
+            all_attributes: Specified property statistics of features
+        """
+
+        #First 103 features are element fractions
+        n_elements = 103
+        element_fractions = [0] * n_elements
+        el_list = list(comp.element_composition.fractional_composition.items())
+        for el in el_list:
+            obj = el
+            atomic_number_i = obj[0].number - 1
+            element_fractions[atomic_number_i] = obj[1]
+
+        #Calculate the remaining features
+        # Get the element names and fractions
+        elements, fractions = zip(*comp.element_composition.items())
+        attributes = [0] * (len(self.features) - n_elements)
+
+        #Total valence electrons, stored so we can calculate orbital fractions
+        valence_sum = 0
```

Comment:
These features are the same as those computed by the [`ValenceOrbital`](https://hackingmaterials.lbl.gov/matminer/matminer.featurizers.html#matminer.featurizers.composition.ValenceOrbital) featurizer. Please make a call out to the existing featurizer rather than making duplicate code.

Reply:
Changed